Add helpful error message for 'uv activate'

nooscraft opened the pull request:
When users try to run 'uv activate', show a helpful error message
instead of the generic unknown command error. The message explains
that 'uv activate' is not supported and provides instructions on
how to activate a virtual environment.

This is most probably address the issue: https://github.com/astral-sh/uv/issues/16993

nooscraft committed:
Add helpful error message for 'uv activate'

When users try to run 'uv activate', show a helpful error message
instead of the generic unknown command error. The message explains
that 'uv activate' is not supported and provides instructions on
how to activate a virtual environment.

This addresses issue #16993.

nooscraft commented:
@charliermarsh @zanieb FYI!

darth-raijin reviewed (commented):
Small suggestion about some possibly redundant comments

darth-raijin commented:
+1 on the above. 

It feels like it could set a precedent for adding similar custom messages for other unsupported commands, instead of handling them in a more general way.

nooscraft committed:
Add shell detection and venv discovery to 'uv activate' error message

- Detect user's current shell (Bash, Zsh, Fish, Nushell, Csh, PowerShell, Cmd)
- Search for virtual environment in current and parent directories
- Provide shell-specific activation command with correct path
- Show helpful hint when no venv is found

Addresses feedback from @zanieb to use shell detection logic
and discover the virtual environment.
